contrib/devtools/check-rpc-mappings.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python3
+'''
+Check RPC argument consistency.
+'''
+import sys, os, re
+from collections import defaultdict
+
+# Source files (relative to root) to scan for dispatch tables
+SOURCES = [
+"src/rpc/server.cpp",
+"src/rpc/blockchain.cpp",
+"src/rpc/mining.cpp",
+"src/rpc/misc.cpp",
+"src/rpc/net.cpp",
+"src/rpc/rawtransaction.cpp",
+"src/wallet/rpcwallet.cpp",
+]
+# Source file (relative to root) containing conversion mapping
+SOURCE_CLIENT = 'src/rpc/client.cpp'
+# Argument names that should be ignored in consistency checks
+IGNORE_DUMMY_ARGS = {'dummy','arg0','arg1','arg2','arg3','arg4','arg5','arg6','arg7','arg8','arg9'}
+
+def parse_string(s):
+    assert(s[0] == '"')
+    assert(s[-1] == '"')
+    return s[1:-1]
+
+def make_string(s):
+    return '"%s"' % s
+
+class RPCCommand:
+    def __init__(self, name, args):
+        self.name = name
+        self.args = args
+
+class RPCArgument:
+    def __init__(self, names, idx):
+        self.names = names
+        self.idx = idx
+
+def process_commands(fname):
+    '''
+    Find and parse dispatch table in implementation file `fname`.
+    '''
+    cmds = []
+    in_rpcs = False
+    with open(fname, "r") as f:
+        for line in f:
+            line = line.rstrip()
+            if not in_rpcs:
+                if re.match("static const CRPCCommand .*\[\] =", line):
+                    in_rpcs = True
+            else:
+                if line.startswith('};'):
+                    in_rpcs = False
+                elif '{' in line and '"' in line:
+                    m = re.search('{ *("[^"]*"), *("[^"]*"), *&([^,]*), *(true|false), *{([^}]*)} *},', line)
+                    if not m:
+                        print('No match: %s' % line)
+                        exit(1)
```

Comment:
According to the Python documentation `exit(…)` should be used in the interactive interpreter shell but not in programs:
> The site module (which is imported automatically during startup, except if the -S command-line option is given) adds several constants to the built-in namespace. They are useful for the interactive interpreter shell and should not be used in programs.

Links:
* [`sys.exit(…)`](https://docs.python.org/3/library/sys.html#sys.exit): _"Exit from Python."_
* [`exit(…)`](https://docs.python.org/3/library/constants.html#exit): _"raise SystemExit with the specified exit code. […] should not be used in programs."_